spec/git-repository-async-spec.js

```diff
@@ -0,0 +1,598 @@
+      repo.destroy()
+
+      let threw = false
+      try {
+        await repo.getShortHead()
+      } catch (e) {
+        threw = true
+      }
+      expect(threw).toBeTruthy()
+    })
+  })
+
+  describe('.getPathStatus(path)', () => {
+    let filePath
+
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+      filePath = path.join(workingDirectory, 'file.txt')
+    })
+
+    it('trigger a status-changed event when the new status differs from the last cached one', async () => {
+      const statusHandler = jasmine.createSpy('statusHandler')
+      repo.onDidChangeStatus(statusHandler)
+      fs.writeFileSync(filePath, '')
+
+      await repo.getPathStatus(filePath)
+
+      expect(statusHandler.callCount).toBe(1)
+      const status = Git.Status.STATUS.WT_MODIFIED
+      expect(statusHandler.argsForCall[0][0]).toEqual({path: filePath, pathStatus: status})
+      fs.writeFileSync(filePath, 'abc')
+
+      await repo.getPathStatus(filePath)
+      expect(statusHandler.callCount).toBe(1)
+    })
+  })
+
+  describe('.getDirectoryStatus(path)', () => {
+    let directoryPath, filePath
+
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+      directoryPath = path.join(workingDirectory, 'dir')
+      filePath = path.join(directoryPath, 'b.txt')
+    })
+
+    it('gets the status based on the files inside the directory', async () => {
+      await repo.checkoutHead(filePath)
+
+      let result = await repo.getDirectoryStatus(directoryPath)
+      expect(repo.isStatusModified(result)).toBe(false)
+
+      fs.writeFileSync(filePath, 'abc')
+
+      result = await repo.getDirectoryStatus(directoryPath)
+      expect(repo.isStatusModified(result)).toBe(true)
+    })
+  })
+
+  describe('.refreshStatus()', () => {
+    let newPath, modifiedPath, cleanPath
+
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+      modifiedPath = path.join(workingDirectory, 'file.txt')
+      newPath = path.join(workingDirectory, 'untracked.txt')
+      cleanPath = path.join(workingDirectory, 'other.txt')
+      fs.writeFileSync(cleanPath, 'Full of text')
+      fs.writeFileSync(newPath, '')
+      newPath = fs.absolute(newPath) // specs could be running under symbol path.
+    })
+
+    it('returns status information for all new and modified files', async () => {
+      fs.writeFileSync(modifiedPath, 'making this path modified')
+      await repo.refreshStatus()
+
+      expect(await repo.getCachedPathStatus(cleanPath)).toBeUndefined()
+      expect(repo.isStatusNew(await repo.getCachedPathStatus(newPath))).toBeTruthy()
+      expect(repo.isStatusModified(await repo.getCachedPathStatus(modifiedPath))).toBeTruthy()
+    })
+  })
+
+  describe('buffer events', () => {
+    let repository
```

Comment:
Super minor, but it looks like most of the other times in other specs this variable is called `repo` and not `repository`, not sure if it would be worth using a consistent name throughout, but just wanted to mention it.